jobs/waiting-for-jobs.rst

```diff
@@ -0,0 +1,216 @@
+
+Here's a simple example of using ``flux job wait``.  It's very similar to the example from before, where run sleep for 30 seconds then run ``/bin/true``.
+
+.. code-block:: console
+
+    $ flux submit --flags waitable -n1 bash -c "sleep 30; /bin/true"
+    ƒ4btMovw
+    $ flux job wait ƒ4btMovw
+    <we wait a little bit waiting for the job to finish>
+    $ echo $?
+    0
+
+Note that when submitting the job, we submitted it with the ``waitable`` flag via ``--flags waitable``.
+
+This doesn't really show us anything special, it seems to be the same as ``flux job status``.  Lets now look at the major advantages of ``flux job wait``.
+
+Perhaps the biggest advantage of ``flux job wait`` is that apriori knowledge of jobids is not necessary.  If ``flux job wait`` is specfied without any jobids, it will wait for the first job that completes amongst all of the jobs you have submitted via the ``waitable`` flag.
+
+.. code-block:: console
+
+    $ flux submit --flags waitable -n1 bash -c "sleep 60; /bin/true"
+    ƒ2WxyXSUF
+    $ flux submit --flags waitable -n1 bash -c "sleep 45; /bin/true"
+    ƒ2XRcLY7Z
+    $ flux submit --flags waitable -n1 bash -c "sleep 30; /bin/true"
+    ƒ2Zjt9VSw
+    $ flux job wait
+    ƒ2Zjt9VSw
+    $ flux job wait
+    ƒ2XRcLY7Z
+    $ flux job wait
+    ƒ2WxyXSUF
+    $ flux job wait
+    flux-job: there are no more waitable jobs
+
+In this above example, we submit three jobs, sleeping for 60, 45, and 30 seconds respectively before running ``/bin/true``.  We then run ``flux job wait`` without any inputs.  You'll notice the jobids for the ``sleep 30`` job, then ``sleep 45`` job, then ``sleep 60`` job are returned in that order.  Finally, without any jobs left running with the ``waitable`` flag, ``flux job wait`` indicates there are no more waitable jobs.
+
+Another option is that all jobs can be waited on via the ``--all`` option to ``flux job wait``.  Lets try that in the below example.
+
+.. code-block:: console
+
+    $ flux submit --flags waitable -n1 bash -c "sleep 60; /bin/true"
+    ƒ4YNPpFmAf
+    $ flux submit --flags waitable -n1 bash -c "sleep 45; /bin/true"
+    ƒ4YPufmCjq
+    $ flux submit --flags waitable -n1 bash -c "sleep 30; /bin/false"
+    ƒ4YSVQWfZq
+    $ flux job wait --all --verbose
+    ƒ4YSVQWfZq: task(s) exited with exit code 1
+    ƒ4YPufmCjq: job completed successfully
+    ƒ4YNPpFmAf: job completed successfully
+    $ echo $?
+    1
+
+This example is similar to the above, except one of the jobs runs ``/bin/false`` instead of ``/bin/true``.  When ``flux job wait --all`` is executed, you'll notice a message output indicating that one job has failed (the one that ran ``/bin/false``).  And similar to ``flux job status``, the exit code of ``1`` is returned due to the highest exit code of all the jobs.
+
+The biggest disadvantage of ``flux job wait`` compared to ``flux job status`` is that jobs can only waited on once.
```

Comment:
```suggestion
The biggest disadvantage of ``flux job wait`` compared to ``flux job status`` is that jobs can only be waited on once.
```

Reply:
Only being able to wait on a job once is not necessarily a disadvantage, without it you would not be able to `flux job wait` in a loop (you'd just keep getting the same jobid continually). So there is a purpose here and each interface satisfies different se cases.  Instead of calling this a disadvantage, maybe the guide should discuss the use cases for which each interface is designed?